lib-js/directory.ts

```diff
@@ -0,0 +1,102 @@
+import yaml from "js-yaml"
+import { promises as fs } from "fs"
+import path from "path"
+
+import { withArchive } from "./hrx"
+
+interface YamlOptions {
+  ":todo"?: string[]
+  ":ignore_for"?: string[]
+  ":warning_todo"?: string[]
+  ":precision"?: number
+}
+
+function hasOptionForImpl(option: string[] | undefined, impl: string) {
```

Comment:
It would help me for reviews to have these private functions/interfaces documented as well.